### ISSUE TITLE
Everyone is blocked from seeing a post a blocked user replied to, or that has blocked them.

### ISSUE THREAD

**Issue by cyphix333** [bluesky-social/social-app#7021]
### Steps to Reproduce

1. Access a post either via a quoted post, or via a reply link directly, or within your feed.
2. The persons post you are viewing should be in reply to someone else. If the persons post you are viewing happens to be blocked by the person they are replying to (or the person replying to them has now blocked them), this now means YOU cannot see that post either, whether you are blocked (or have blocked them) or not.

Initially I was seeing this a lot and just figured I was blocked by the person, but I confirmed this was not the case with someone that was in the same thread as me. The person told me they had blocked the person, and I could see this person's posts they blocked fine, UNLESS I was viewing a post made by the person that blocked them (I could not see the post they were replying to).

So I guess I have confirmed it one way (the person blocked them), but I suspect it works both ways (if they are blocked).

### What platform(s) does this occur on?

Android, Web (Desktop)

### What version of the app are you using?

1.95.0.634

**Comment by cooljeanius** [comment]
This is by design and why BlueSky's block is known as the "nuclear block". I do agree that it's kind of confusing, though, and could probably be documented better.

**Comment by cyphix333** [comment]
> This is by design and why BlueSky's block is known as the "nuclear block". I do agree that it's kind of confusing, though, and could probably be documented better.

I understand the need for a strong block function, but this is essentially letting other people decide what content I see. So, if someone else I'm in a thread with has blocked someone, then that makes ME (and everyone else) not able to see the content of someone that THEY decided to block, as well as the resulting thread above it.

Not good at all.

**Comment by skyduct** [comment]
closed with WONTFIX - system is working as designed.
wait, not closed, because I'm not a bsky dev and can't close it.

might be nice for it to say who blocked who, though.

**Comment by diesocke** [comment]
It is a feature, not a bug.

**Comment by cyphix333** [comment]
> It is a feature, not a bug.

Yes, I already realise that, but this "feature" makes the user experience of that functionality of the site awful and really doesn't help anybody.

**Comment by Tamschi** [comment]
It's a very popular feature, as it makes it largely useless to troll or harass for attention.

Though, Bluesky should go a bit further and not account for these replies in the reply counter.

**Comment by hutajoullach** [comment]
I thought the nuclear block would hide posts, replies, mentions, and profiles of blocked accounts.

Not sure if the reply counter will subtract the number of blocked accounts for that particular user though.

**Comment by cyphix333** [comment]
> It's a very popular feature, as it makes it largely useless to troll or harass for attention.
> 
> Though, Bluesky should go a bit further and not account for these replies in the reply counter.

It's a popular feature to allow others to control what you see in certain situations? You seem to be referencing the general strong blocking feature based on your comment of trolling and harassment; how does me not being able to see the content of what someone else replied to help with that? It doesn't - instead it removes all context from their response, and makes it harder for you to be able to read or engage in the thread.

Outside of someone blocking ME, I don't want others to control what I can and can't see.

**Comment by diesocke** [comment]
If there is dispute that leads at least one participant to block someone, Bluesky reduces the trolling and escalation potential other platforms have demonstrated to the quite enormous detriment of their usability.

**Comment by cyphix333** [comment]
> If there is dispute that leads at least one participant to block someone, Bluesky reduces the trolling and escalation potential other platforms have demonstrated to the quite enormous detriment of their usability.

I'm sorry, but that tiny benefit isn't a good enough reason. 

As it stands, if 20 people are involved in a discussion, and ONE single person decides to block someone, then all of a sudden, the 19 other people in the discussion (+ any other viewers) are now inconvenienced simply because one person had an issue with someone else.

BS does have a bit of a block culture, and as such, this issue is only going to get worse and worse, and threads are going to get harder and harder to read and follow as more and more people get blocked.

Just the other day I got a notification, and I clicked on it, and once again, the post they were replying to was "blocked", not because of me, but because the person who made the post blocked the person they were responding to. I was trying to make sense of their post, but now I couldn't as I had no idea what the hell they were replying to... then I *think* I found the post they replied to; it showed "1 reply", but when I clicked on it, no replies were shown.

Now, this functionality was probably done with good intentions - but you know what they say, "The road to hell is paved with good intentions".

**Comment by ppoutine** [comment]
This is working as intended but i agree it should be reassessed. For example:

1. In a popular thread, User A posts some nonsense
2. User B replies to that reply explaining why it's nonsense
3. User A blocks User B
4. Now User A has successfully hidden the rebuttal to his comment from everyone. The only defense against this is if the thread OP happens to block User A.

This is a pretty serious downside of the "nuclear block" system imo. It creates an escalation ladder of blocking where the first user to hit "block" is advantaged. On the other hand it causes me personally to _avoid_ blocking where I otherwise would, because I want the conversation to still be visible for others.

There should at least be a "show reply" button on posts that are hidden for this reason IMO. Otherwise you've given every user the unilateral power to hide a reply, for everyone, permanently. If you hide a reply the normal way, it's not deleted for everyone! There is a "show hidden reply" button! The effect of hiding someone else's reply should be consistent across these two ways to do it.

**Comment by cyphix333** [comment]
> This is a pretty serious downside of the "nuclear block" system imo. It creates an escalation ladder of blocking where the first user to hit "block" is advantaged. On the other hand it causes me personally to _avoid_ blocking where I otherwise would, because I want the conversation to still be visible for others.

Some very good points, and show multiple other sides to the issue.

I get the idea about this functionality, but the cascading effects into other areas is too much; it's essentially throwing the baby out with the bathwater.

**Comment by skyduct** [comment]
perhaps there could be an additional block type which is "non retroactive".

**Comment by evecodex** [comment]
I'm guessing this is the weirdness I'm seeing then. If so, then this has reached the point of being problem. Twice this morning I went to look at notifications from one of my followers from a post they created, but couldn't see the thread because it had a "Blocked post" message at the top. I can view my follower's reply from my Notifications list, but I only get a message "Blocked post" if I go to the thread directly. 

The second instance I had of this was on my own post. Someone replied to my post, but I couldn't view my own thread I'm guessing because someone in MY thread was blocking someone else in the thread. 

I personally only use one blocklist (for bots that is maintained regularly by the dev) to prevent this problem. But other users who aren't as savvy may just blindly follow blocklists from other "devs" who are using them as revenge blacklists to kill another user's engagement or maybe they're just doing it to sabotage Bluesky in general. All this blocking going on is causing people to see "ghost" replies where I see someone has replied but there's no one in the thread, ghost likes, "blocked posts" and all kinds of other "anomalies". I saw another user complaining about ghost likes in her feed as well and she said she's seeing the number of ghost engagements increasing. 

Also, I've noticed I was instantly added to a massive blocklist for blocking the creator of these known "revenge" blocklists. I guess their list is automated because I was added to their blocklist within minutes after blocking their account. There is far too much blocking going on which is starting to render the platform to looking buggy. I don't think you guys realize how large some of these blocklists are. I'm not going to drop the user's info here, but there is one user who is abusing blocklists, running multiple block lists and just one of those lists is blocking over 60,000 accounts. It's lists like this that are ruining the platform. 

You have two situations clashing against each other and it's ruining peoples' experience. You can keep this charade up if you want to, but it's going to eventually start spreading and more people are going to complain.

**Comment by DHager** [comment]
> 4. Now User A has successfully hidden the rebuttal to his comment from everyone. The only defense against this is if the thread OP happens to block User A.

Worse:  User A can use this strategy to effectively lock user B out of the entire conversation, even if the grandparent-poster and 99% of the other people involved are quite happy to interact with B.

This occurs because User B, viewing their own reply-history, can no navigate "up past" User A's content to reach the main discussion again. (Or at least, I can't find any way to do it in the Android app.)

**Comment by cyphix333** [comment]
> Worse: User A can sometimes use this strategy to effectively lock User B out of the _entire_ conversation, even when literally everybody else (including the top-level post-author) are perfectly happy with User B's participation.
> 
> This occurs when User B has made just one reply so far, and then later they try to return but can't navigate "up past" User A's mutually-blocked content to re-acquire the main discussion again. (Or at least, I can't find any way to do it in the Android app.) In effect, User A has unilaterally sabotaged User B's bookmarks / history, barring them from the entire conversation unless they can remember the right usernames/keywords to search for it again.

Yep, so many problematic scenarios with the way it currently works. As I said, it was likely put in with good intentions, but there is just far far too many adverse scenarios that makes navigating/participating in the thread impossible, or extremely frustrating for everyone else once a SINGLE user makes the unilateral decision to block someone else.

**Comment by AngerySnek** [comment]
> It's a very popular feature, as it makes it largely useless to troll or harass for attention.
> 
> Though, Bluesky should go a bit further and not account for these replies in the reply counter.

If I make a comment on a social site, i expect to be able to see the full thread of whatever topic i'm viewing. If some idiot just blocked me or someone else for no reason, then i'm shit out of luck there. There's no way this is a good feature by any stretch as it can also be abused by trolls as anyone can block willy nilly, thus hiding every comment of theirs on connected from being seen

At least twitter had hiding comments, which could be expanded so that anyone under your posts/comments could be hidden regardless of OP's intentions.

**Comment by Magedeline** [comment]
hey there I want to ask about this and I have been thinking about using the new feature to blocked new user from my blue sky account is there by anychances for me to use it?

**Comment by witcheslive** [comment]
This is still a ridiculous mal-feature that has been widely abused and done nothing but cause confusion and form weird cults all around the site. More often than not, a blocker is an antisocial troll that blocks because someone replying to them was correct in some way, and they want to hide this pushback, and control information. There is already the ridiculous but at least understandable feature that lets anyone disable replies or quotes that is perfect for people who don't want to be challenged or directly interacted with that can be applied as necessary, so why give trolls and people with anti-social positions the ability to effectively delete others' posts to further their anti-social agendas? 

We've been putting up with this ridiculous mal-feature for over a year now, there's another microblogging platform for control freaks that want to dictate a right-wing reality to their thralls, bluesky should be the place where you don't have to have right wing messaging shoved down your maw without the ability to comment or respond.

**Comment by Magedeline** [comment]
I mean yeah but heres the point I get it that few people like them doesn't want me to to talk to them nor go to the alt account just because the alt account that I find is good, for example: trolls who they impersonated me or insulted or make false allegation about me they will be flagged as impersonator meaning bluesky knows that they came to harrassed someone or somebody or me, so what I do? I blocked them of course, but if they made a new account and see my user they still get blocked meaning alts account are not welcome to my main profile or anyone really.

so thats why bluesky has added that feature

**Comment by AngerySnek** [comment]
> I mean yeah but heres the point I get it that few people like them doesn't want me to to talk to them nor go to the alt account just because the alt account that I find is good, for example: trolls who they impersonated me or insulted or make false allegation about me they will be flagged as impersonator meaning bluesky knows that they came to harrassed someone or somebody or me, so what I do? I blocked them of course, but if they made a new account and see my user they still get blocked meaning alts account are not welcome to my main profile or anyone really.
> 
> so thats why bluesky has added that feature

Is that english? Your comment is just so full of filler text, I can hardly make sense of it. This feature sucks and Bluesky needs to change it to at least be optional. No ifs, ands, or buts.

**Comment by Magedeline** [comment]
maybe but I ain't invovled with this

**Comment by ghost** [comment]
Hi. I noticed this weird behavior and also assumed that it must be a bug. If this is a feature, then I honestly don't understand its purpose, because it is obviously abused for censorship. I understand that if person A blocks person B, then person A is the one who shouldn't (because they obviously don't want to) see responses of person B, but if person A can basically directly hide responses of other people for **everybody else**, then it seems to me like "individual censorship". The problem here is: Person A might not like what person B replies for whatever reason (which is absolutely ok), but this doesn't necessarily apply to other users on the platform who would probably like to see the response of person B. Person A doesn't have to justify their decision if they are blocking someone elses post for themself personally, but if they have the power to make this decision for everybody else, this doesn't seem to be justified from a logical standpoint. The argument with anti trolling and harassing cannot be applied generally, because everyone can block anyone for any reason - even just for having a different opinion, which seems to happen quite often. This leads to personal, opinion based "discussion bubbles" (including spreading misinformation, because "unwanted" replies can be hidden by the person who is spreading misinformation) where only opinions are accepted and visible who you agree with, rather than an open platform for everybody to participate in. It makes discussions impossible and often breaks existing discussion retrospectively.

In short: Others control what you see. That doesn't seem to be a good feature. The disadvantages seem to outweight the advantages in my opinion.

**Comment by cyphix333** [comment]
It would be nice of this issue was at least addressed by someone from BlueSky - it's obvious a lot of people find this frustrating, so ignoring it isn't just going to make it go away.

**Comment by cooljeanius** [comment]
So, unfortunately the ctxplz bot got banned, but I'd just like to make sure that people on this issue are aware that the DMs-only alternative to it still exists: https://bsky.app/profile/ctxpsst.bsky.social

**Comment by cyphix333** [comment]
> So, unfortunately the ctxplz bot got banned, but I'd just like to make sure that people on this issue are aware that the DMs-only alternative to it still exists: https://bsky.app/profile/ctxpsst.bsky.social

Did you post to the wrong issue?

**Comment by cooljeanius** [comment]
> > So, unfortunately the ctxplz bot got banned, but I'd just like to make sure that people on this issue are aware that the DMs-only alternative to it still exists: [bsky.app/profile/ctxpsst.bsky.social](https://bsky.app/profile/ctxpsst.bsky.social)
> 
> Did you post to the wrong issue?

No. The ctx bots were created as a response to the problem described in this issue.

**Comment by fractalman237** [comment]
> closed with WONTFIX - system is working as designed. wait, not closed, because I'm not a bsky dev and can't close it.
> 
> might be nice for it to say who blocked who, though.

Billionaires screwing the bottom 90% is a feature of rampant capitalism; not a bug. Does that mean that we shouldn't try to fix it?
